README.md
<img src="http://opendatadpc.maps.arcgis.com/sharing/rest/content/items/5c8ef7516b5b4bb19f61037b4cd69015/data" alt="COVID-19" data-canonical-src="http://opendatadpc.maps.arcgis.com/sharing/rest/content/items/5c8ef7516b5b4bb19f61037b4cd69015/data" width="400" />

[Italiano](README.md) - [English](README_EN.md)<br><br>

# Dati COVID-19 Italia

[![GitHub license](https://img.shields.io/badge/License-Creative%20Commons%20Attribution%204.0%20International-blue)](https://github.com/pcm-dpc/COVID-19/blob/master/LICENSE)
[![GitHub commit](https://img.shields.io/github/last-commit/pcm-dpc/COVID-19)](https://github.com/pcm-dpc/COVID-19/commits/master)
 
<<<<<<< HEAD
=======
## Avviso modifiche 
[https://github.com/pcm-dpc/COVID-19/issues/625](https://github.com/pcm-dpc/COVID-19/issues/625)<br><br>
>>>>>>> c986b32ec52c99a6105af6224b4d5ead6d94c71e

[Sito del Dipartimento della Protezione Civile - Emergenza Coronavirus: la risposta nazionale](http://www.protezionecivile.it/attivita-rischi/rischio-sanitario/emergenze/coronavirus)


Il 31 gennaio 2020, il Consiglio dei Ministri dichiara lo stato di emergenza, per la durata di sei mesi, in conseguenza del rischio sanitario connesso all'infezione da Coronavirus.

Al Capo del Dipartimento della Protezione Civile, Angelo Borrelli, è affidato il coordinamento degli interventi necessari a fronteggiare l'emergenza sul territorio nazionale.  
  
Le principali azioni coordinate dal Capo del Dipartimento sono volte al soccorso e all'assistenza della popolazione eventualmente interessata dal contagio, al potenziamento dei controlli nelle aree aeroportuali e portuali, in continuità con le misure urgenti già adottate dal Ministero della salute, al rientro in Italia dei cittadini che si trovano nei Paesi a rischio e al rimpatrio dei cittadini stranieri nei Paesi di origine esposti al rischio.

Per informare i cittadini e mettere a disposizione i dati raccolti, utili ai soli fini comunicativi e di informazione, il Dipartimento della Protezione Civile ha elaborato un cruscotto geografico interattivo raggiungibile agli indirizzi  [http://arcg.is/C1unv](http://arcg.is/C1unv) (versione desktop) e [http://arcg.is/081a51](http://arcg.is/081a51) (versione mobile) e mette a disposizione, con licenza CC-BY-4.0, le seguenti informazioni aggiornate quotidianamente alle 18:30 (successivamente la conferenza stampa del Capo Dipartimento):

- Dati Andamento nazionale
- Dati json
- Dati regioni
- Dati province
- Schede riepilogative
- Aree
- Note
- Dati contratti DPC forniture

## Avvisi

[Avvisi sui dati andamento COVID-19 Italia](avvisi.md)<br>

## Struttura del repository
```
COVID-19/
│
├── aree/
│   ├── geojson
│   │   ├── dpc-covid-19-ita-aree-comuni.geojson
│   │   ├── dpc-covid19-ita-aree.geojson
│   ├── shp
│   │   ├── dpc-covid19-ita-aree-comuni.dbf
│   │   ├── dpc-covid19-ita-aree-comuni.prj
│   │   ├── dpc-covid19-ita-aree-comuni.shp
│   │   ├── dpc-covid19-ita-aree-comuni.shx
│   │   ├── dpc-covid19-ita-aree.dbf
│   │   ├── dpc-covid19-ita-aree.prj
│   │   ├── dpc-covid19-ita-aree.shp
│   │   ├── dpc-covid19-ita-aree.shx
├── dati-andamento-nazionale/
│   ├── dpc-covid19-ita-andamento-nazionale-*.csv
│   ├── dpc-covid19-ita-andamento-nazionale-latest.csv
│   ├── dpc-covid19-ita-andamento-nazionale.csv
├── dati-contratti-dpc-forniture/
│   ├── dpc-covid19-dati-contratti-dpc-forniture.csv
│   ├── dpc-covid19-dati-pagamenti-contratti-dpc-forniture.csv
│   ├── dati-json
│   │   ├── dpc-covid19-dati-contratti-dpc-forniture.csv
│   │   ├── dpc-covid19-dati-pagamenti-contratti-dpc-forniture.csv
│   ├── file-atti-negoziali
│   │   ├── dpc-contratto-covid19-*.pdf
├── dati-json/
│   ├── dpc-covid19-ita-andamento-nazionale-latest.json
│   ├── dpc-covid19-ita-andamento-nazionale.json
│   ├── dpc-covid19-ita-note-en.json
│   ├── dpc-covid19-ita-note-it.json
│   ├── dpc-covid19-ita-province-latest.json
│   ├── dpc-covid19-ita-province.json
│   ├── dpc-covid19-ita-regioni-latest.json
│   ├── dpc-covid19-ita-regioni.json
├── dati-province/
│   ├── dpc-covid19-ita-province-*.csv
│   ├── dpc-covid19-ita-province-latest.csv
│   ├── dpc-covid19-ita-province.csv
├── dati-regioni/
│   ├── dpc-covid19-ita-regioni-*.csv
│   ├── dpc-covid19-ita-regioni-latest.csv
│   ├── dpc-covid19-ita-regioni.csv
├── note/
│   ├── dpc-covid19-ita-note-en.csv
│   ├── dpc-covid19-ita-note-it.csv
├── schede-riepilogative/
│   ├── province
│   │   ├── dpc-covid19-ita-scheda-province-*.pdf
│   ├── regioni
│   │   ├── dpc-covid19-ita-scheda-regioni-*.pdf
```

## Aggiornamento dei dati

- Dati andamento COVID-19 Italia: ogni giorno alle 18:00<br>
- Dati contratti DPC COVID-19 di fornitura: continua (ogni volta che vengono effettuate operazioni sui contratti)

## Formato dei dati

- [Dati andamento COVID-19 Italia](dati-andamento-covid19-italia.md)<br>
- [Dati contratti DPC COVID-19 di fornitura](dati-contratti-dpc-covid19-fornitura.md)
- [Dati aree misure restrittive COVID19](dati_aree-nazionali-subregionali-misure-restrittive-covid19.md)

## Licenza

[CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/deed.it) - [Visualizza licenza](https://github.com/pcm-dpc/COVID-19/blob/master/LICENSE)


CHANGELOG.md
# Changelog

Tutte le modifiche al progetto, nuove funzionalità e informazioni sono documentate in questo file

## 2020-03-08

- Modifica: "dati-andamento-nazionale" riportando i totali dei dati delle Regioni
- Rimossa: directory "shape-aree-contenimento" e creata "aree" con "shp" e "geojson"

## 2020-03-10

- Aggiornate le Aree secondo il nuovo DPCM 9 Marzo 2020

## 2020-03-11

- Modifica: "Bolzano" e "Trento" rinominati in "P.A. Bolzano" e "P.A. Trento" in "dati-regioni e "dati-province" ("denominazione_regione")
- Modifica: Friuli V. G. rinominata in "dati-regioni e "dati-province" ("denominazione_regione") in "Friuli Venezia Giulia"

## 2020-03-12

- Aggiunto: Codice di Condotta in italiano (CODE_OF_CONDUCT.md) e in inglese (CODE_OF_CONDUCT_EN.md)
- Aggiunto: Changelog in italiano (CHANGELOG.md) e in inglese (CHANGELOG_EN.md)
- Aggiornate le Aree secondo il nuovo DPCM 11 Marzo 2020

## 2020-03-19

- Aggiunti ultimi dati (latest) csv, nelle rispettive directory, per andamento nazionale (dpc-covid19-ita-andamento-nazionale-latest.csv), regioni (dpc-covid19-ita-regioni-latest.csv) e province (dpc-covid19-ita-province-latest.csv)
- Aggiunti ultimi dati (latest) json, nella directory dati-json, per andamento nazionale (dpc-covid19-ita-andamento-nazionale-latest.json), regioni (dpc-covid19-ita-regioni-latest.json) e province (dpc-covid19-ita-province-latest.json)

## 2020-03-25

- Data nel formato ISO8601 UTC
- Aggiunta: "Note" in "dati-regioni", "dati-province" e "dati-andamento-nazionale"
- Aggiunta: dataset "note"

## 2020-03-30

- Modifica: "totale_attualmente_positivi" rinominato in "totale_positivi" (ricoverati_con_sintomi + terapia_intensiva + isolamento domiciliare) in "dati_regioni" e "dati_andamento_nazionale"
- Modifica: "nuovi_attualmente_positivi" rinominato in "variazione_totale_positivi" (totale_attualmente positivi giorno corrente - totale_attualmente positivi giorno precedente) in "dati_regioni" e "dati_andamento_nazionale"
- Aggiunta: "nuovi_positivi" (totale_casi giorno corrente - totale_casi giorno precedente) in "dati_regioni" e "dati_andamento_nazionale"
- Modifica: Regione "Emilia Romagna" rinominato in "Emilia-Romagna" in "dati-regioni" e "dati-province" ("denominazione_regione")

## 2020-04-20

- Aggiunta: "casi_testati", totale dei soggetti sottoposti al test dal 19/04/2020

## 2020-04-21

<<<<<<< HEAD
- Aggiunta: nuovo shape file con misure di contenimento a livello subregionale (province, comuni e frazioni).
=======
- Aggiunta: nuovo shape file con misure di contenimento a livello subregionale (province, comuni e frazioni) con rlativi metadata.
>>>>>>> c986b32ec52c99a6105af6224b4d5ead6d94c71e

## 2020-04-22

- Aggiunta: nuovo dataset contratti dpc forniture (contratti e pagamenti) in formato csv e json.  [link alla dashboard](https://app.powerbi.com/view?r=eyJrIjoiNzA1YmQ0NDctYzY3ZC00ZTlkLWI1MTQtMThmNTEwNWY3NjM4IiwidCI6IjlhMDZhOTA3LTA2OTUtNDA0YS05NmY4LTRhMWU5YTJmYjQxZCIsImMiOjl9) - [link al dataset ](https://github.com/pcm-dpc/COVID-19/tree/master/dati-contratti-dpc-forniture)

## 2020-04-23

- Aggiunta: metadati del nuovo dataset contratti dpc forniture (contratti e pagamenti) in formato DCAT-AP-IT
- Integrato README con specifiche per il data model dei dataset aree (nazionali e subregionali)

<<<<<<< HEAD
## Prossimi aggiornamenti

- API Rest (Json)
- GraphQL

## Ipotesi modifiche
=======
## 2020-04-27

- Aggiunta: aggiornamento al 26/4/2020 shape file e relativi metadata con misure di contenimento a livello subregionale (province, comuni e frazioni).

## 2020-05-04

- Aggiunta: aggiornamento al 4/5/2020 shape file e relativi metadati con misure di contenimento a livello nazionale a seguito del DPCM 26/4/2020.

## 2020-05-05

- API Rest (Json) private beta
- GraphQL private beta

## 2020-05-06

- Aggiunta: aggiornamento al 03/05/2020 shape file e relativi metadata con misure di contenimento a livello subregionale (province, comuni e frazioni).

## Prossimi aggiornamenti
>>>>>>> c986b32ec52c99a6105af6224b4d5ead6d94c71e

- Cambio codice_regione P.A. Bolzano e P.A. Trento: da 04 a 21 per P.A. Bolzano e da 04 a 22 per P.A. Trento (codice provincia)


script/001_main.R
library(tidyverse)
library(fs)
library(ggthemes)
library(ggrepel)
library(ggsci)
# Font
library(showtext)
# Google fonts: https://fonts.google.com/
font_add("roboto", "~/Downloads/Spartan/static/Spartan-ExtraLight.ttf")
font_add("roboto2", "~/Downloads/Spartan/static/Spartan-Regular.ttf")
showtext_auto()
## turn off if no longer needed
showtext_auto(FALSE)

file <- function(){
file_info(dir_ls()) %>%
select(path, type, size, birth_time, change_time, modification_time, access_time) %>%
arrange(desc(modification_time))}


################################################################################
# Read data procince
f <- dir_ls("../dati-province/", recurse = F, regex = "\\d\\.csv$")

x <- map_df(f, read_csv)

x %>% 
    filter(denominazione_regione == "Lombardia") %>%
#     filter(denominazione_provincia == "Torino",
#            !grepl("2020-03-19", data)) %>%
    ggplot(aes(data, totale_casi)) +
    geom_point() +
    geom_smooth() +
    theme_economist() +
    facet_wrap(~denominazione_provincia)


script/002_Tracking.R
library(tidyverse)
library(here)
library(lubridate)
library(sf)
library(gganimate) #Used for animations
library(RColorBrewer) #Used for color scale 

setwd(here())

# Read data
covid19_data <- read_csv("dati-province/dpc-covid19-ita-province.csv")  %>%
    select(data, DP = denominazione_provincia, totale_casi, lat, long) %>%
    mutate(data = ymd_hms(data))

# Filter data
co <- covid19_data %>%
    filter(DP != "In fase di definizione/aggiornamento")

# https://www.diva-gis.org/datadown
it <- st_read("Italy_shapefile/ITA_adm/")

# Join
sp <- co %>%
    left_join(it, by = c("DP" = "NAME_2")) %>%
    select(data:totale_casi, geometry)

sp <- st_as_sf(sp) %>%
    mutate(data = as.character(data), 
           data = str_extract(data, "\\d\\d\\d\\d-\\d\\d-\\d\\d"), 
           data = ymd(data))

################################################################################
# Plot
# Makes plot with ggplot2 and gganimate to animate through the days 
sp_map<-ggplot()+
  geom_sf(data = it,fill = "white")+
  geom_sf(data = sp %>%filter(data %in% c(ymd("2020-02-24"), ymd("2020-03-01"))),aes(fill=totale_casi))+
  ggtitle("Spread of Covid19 Throughout Italy")+
  xlab("")+
  ylab("")+
  labs(subtitle = "Date: {current_frame}",
       caption = "Date Source: The New York Times\nAuthor: Kevin Zolea")+
  cowplot::background_grid(major = "none", minor = "none") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        axis.line = element_blank(),
        legend.background = element_blank(),
        legend.position=c(-0.3,0.8),
        plot.background = element_blank(),
        panel.background = element_blank(),
        legend.text = element_text(size=12),
        legend.title = element_text(colour="black", size=12, face="bold"),
        plot.title=element_text(size=20, face="bold",hjust =0.5),
        plot.subtitle = element_text(hjust = 0.5,size=12),
        plot.caption = element_text(size = 11,
                                    hjust = .5,
                                    color = "black",
                                    face = "bold"))+
  scale_fill_distiller("Number of Positive Cases",
                       palette ="Reds",type = "div",
                       direction = 1)+
  transition_manual(data)

#   animate(sp_map, nframe=25,fps = 2, end_pause = 15,height = 500, width =500)
  animate(sp_map, renderer = ffmpeg_renderer())




script/003_region.R
library(tidyverse)
library(here)
library(lubridate)
library(sf)
library(gganimate) #Used for animations
library(RColorBrewer) #Used for color scale 
library(ggrepel)

setwd(here())

# Read data
covid19_data <- read_csv("dati-province/dpc-covid19-ita-province.csv") %>%
    select(data, DP = denominazione_provincia, totale_casi, lat, long) %>%
    mutate(data = ymd_hms(data))

# Filter data
co <- covid19_data %>%
    filter(DP != "In fase di definizione/aggiornamento")

# https://www.diva-gis.org/datadown
pr <- st_read("ITA_adm/provincie/")

# Join
sp <- co %>%
    left_join(pr, by = c("DP" = "NAME_2")) %>%
    select(data:totale_casi,NAME_1,  geometry) %>%
    filter(NAME_1 %in% c("Piemonte", "Lombardia"))

sp <- st_as_sf(sp) %>%
    mutate(data = as.character(data), 
           data = str_extract(data, "\\d\\d\\d\\d-\\d\\d-\\d\\d"), 
           data = ymd(data))

################################################################################
# Plot
# Makes plot with ggplot2 and gganimate to animate through the days 
sp_map<-sp %>%
    ggplot(aes(label = NAME_1))+
#   geom_sf(data = it,fill = "white")+
  geom_sf(aes(fill=totale_casi))+
  ggtitle("Spread of Covid19 Throughout Italy")+
  xlab("")+
  ylab("")+
  labs(subtitle = "Date: {current_frame}",
       caption = "Date Source: https://github.com/pcm-dpc/COVID-19\nAuthor: @mchiapello")+
  cowplot::background_grid(major = "none", minor = "none") +
  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),
        axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        axis.line = element_blank(),
        legend.background = element_blank(),
        legend.position="bottom",
        plot.background = element_blank(),
        panel.background = element_blank(),
        legend.text = element_text(size=12),
        legend.title = element_text(colour="black", size=12, face="bold"),
        plot.title=element_text(size=20, face="bold",hjust =0.5),
        plot.subtitle = element_text(hjust = 0.5,size=12),
        plot.caption = element_text(size = 11,
                                    hjust = .5,
                                    color = "black",
                                    face = "bold"))+
  scale_fill_distiller("Number of Positive Cases",
                       palette ="Reds",type = "div",
                       direction = 1)+
  transition_manual(data)

#   animate(sp_map, nframe=25,fps = 2, end_pause = 15,height = 500, width =500)
  animate(sp_map)

################################################################################
library(ggthemes)
library(ggsci)
a <- read_csv("dati-regioni/dpc-covid19-ita-regioni.csv") %>%
    rename(DP = denominazione_regione)

# Piemonte
a %>%
    filter(DP == "Piemonte") %>%
    select(data, DP, ricoverati_con_sintomi:casi_testati) %>%
    pivot_longer(cols = ricoverati_con_sintomi:casi_testati,
                 names_to = "feature",
                 values_to = "num") %>%
    filter(feature %in% c("dimessi_guariti", "deceduti")) %>%
    ggplot(aes(x = as.Date(data), y = num, color = feature)) +
    geom_point() +
    geom_line() +
    theme_fivethirtyeight() +
    scale_colour_uchicago() +
    geom_vline(xintercept = .1)
